pkg/kubelet/dockershim/remote/docker_service.go

```diff
@@ -138,19 +139,32 @@ func (d *dockerService) ContainerStatus(ctx context.Context, r *runtimeApi.Conta
 }
 
 func (d *dockerService) ExecSync(ctx context.Context, r *runtimeApi.ExecSyncRequest) (*runtimeApi.ExecSyncResponse, error) {
-	return nil, fmt.Errorf("not implemented")
+	stdout, stderr, err := d.runtimeService.ExecSync(r.GetContainerId(), r.GetCmd(), time.Duration(r.GetTimeout())*time.Second)
```

Comment:
Yes, `ExecSync` is the non-interactive exec, used internally for probes and health checks (the name isn't great, but I couldn't come up with anything better). `Exec` is the streaming (interactive) version, which does include an stdin stream.